packages/mui-base/src/Button/Button.tsx

```diff
@@ -97,14 +98,7 @@ Button.propTypes /* remove-proptypes */ = {
   /**
    * A ref for imperative actions. It currently only supports `focusVisible()` action.
    */
-  action: PropTypes.oneOfType([
```

Comment:
**Off-topic**

Maybe for API consistency?

```suggestion
  actionRef: PropTypes.oneOfType([
```

Reply:
I don't think it's an improvement - we lost information about the shape of the ref here. The `refType` should apply only to refs to HTML elements IMO.

As a side note, if I remember correctly, we're using `action` props to support `focusVisible` action, which I'd like to drop around v6 (browser support should be sufficient then).
In other cases (like menus, select, etc.) we expose the `dispatch` prop that accepts a function instead.

Reply:
https://github.com/mui/material-ui/pull/38917#discussion_r1321389952

I think the right discussion is here

I think we can close the PR
I did a proposal that could be better than the current state, but I don't think the effort is worth it  for now